src/readfish/plugins/mappy.py

```diff
@@ -43,6 +43,20 @@ def __init__(
         self.logger = setup_debug_logger(__name__, log_file=debug_log)
         self.aligner_params = kwargs
         self.aligner = mappy.Aligner(**self.aligner_params)  # type: ignore
+        if not self.initialised:
```

Comment:
This can be simplified down to one `if/else`

```python
if "fn_idx_in" in self.aligner_params and not Path(self.aligner_params["fn_idx_in"]).is_file():
    raise FileNotFoundError(...)
```

Do we raise to `RunTImeError` as we may not have a reference in some use cases

Reply:
Should the code proceed if no reference is provided? If we have no reference, we should not be initialising the aligner?